source/docs/software/wpilib-tools/robot-characterization/analyzing-feedback.rst

```diff
@@ -0,0 +1,81 @@
+- **Max Controller Output:** This is the maximum value of the controller output, with respect to the PID calculation. Most controllers calculate outputs with a maximum value of 1, but early versions of the Talon firmware have a maximum output of 1023.
+- **Time-Normalized Controller:** This specifies whether the PID calculation is normalized to the period of execution, which affects the scaling of the D gain.
+- **Controller Type:** This specifies whether the controller is an onboard RIO loop, or is running on a smart motor controller such as a Talon or a SPARK Max.
+- **Post-Encoder Gearing:** This specifies the gearing between the encoder and the mechanism itself. This is necessary for control loops that do not allow user-specified unit scaling in their PID computations (e.g. those running on Talons). This will be disabled if not relevant.
+- **Encoder PPR:** This specifies the pulses-per-revolution (not cycles per revolution) of the encoder used, which is needed in the same caes as Post-Encoder Gearing.
+- **Has Slave:** Whether there is a motor controller slaved to the controller running the control loop, if the control loop is being run on a peripheral device. This changes the effective loop period.
+- **Slave Update Period:** The rate at which the slave (if present) is updated. By default, this is 100Hz (every 0.01s) for Talons and Sparks, but can be changed.
+
+Specify Optimality Criteria
+---------------------------
+
+Finally, the user must specify what will be considered an "optimal" controller.  This takes the form of desired tolerances for the system error and control effort - note that it is *not* guaranteed that the system will obey these tolerances at all times.
+
+.. image:: images/optimalitycriteria.png
+   :alt: Entering optimality criteria for the feedback controller
+
+As a rule, smaller values for the ``Max Acceptable Error`` and larger values for the ``Max Acceptable Control Effort`` will result in larger gains - this will result in larger control efforts, which can grant better setpoint-tracking but may cause more violent behavior and greater wear on components.
+
+The ``Max Acceptable Control Effort`` should never exceed 12V, as that corresponds to full battery voltage, and ideally should be somewhat lower than this.
+
+Select Loop Type (Drive Only)
+-----------------------------
+
+It is typical to control drives with both position and velocity PIDs, depending on application. Either can be selected using the drop-down ``Loop Type`` menu.
+
+.. image:: images/looptype.png
+   :alt: Specifying characterization loop type
+
+Enter Known Velocity/Acceleration
+---------------------------------
+
+If one wishes to use the ``Feedback Analysis`` pane without running a full analysis on a set of data, or otherwise view the effect of modifying the ``kV`` and ``kA`` values, this can be done here.
```

Comment:
@AustinShalit got a negative value for ka, but it was really small. We should mention that in that case, overriding that with a ka of 0 is probably fine and will produce similar feedback gains. Furthermore, kp and kd of 0 in that case for a velocity controller are ok in that case because acceleration is negligible; feedforward alone should be adequate.